File: src/main/java/com/core/base/corebase/config/AuthenticationFacade.java
package com.core.base.corebase.config;

import com.core.base.corebase.common.code.ErrorCode;
import com.core.base.corebase.common.exception.BaseException;
import org.springframework.stereotype.Component;
import org.springframework.web.context.annotation.RequestScope;

import java.util.UUID;

@Component
@RequestScope
public class AuthenticationFacade {

    private UUID uid;
    private String email;
    private String name;
    private boolean initialized = false;

    public UUID getUid() {
        if (!initialized) throw new BaseException(ErrorCode.INVALID_TOKEN);
        return uid;
    }

    public String getEmail() {
        if (!initialized) throw new BaseException(ErrorCode.INVALID_TOKEN);
        return email;
    }

    public String getName() {
        if (!initialized) throw new BaseException(ErrorCode.INVALID_TOKEN);
        return name;
    }

    public void setInfo(UUID uid, String email, String name) {
        this.uid = uid;
        this.email = email;
        this.name = name;
        this.initialized = true;
    }
}

File: src/main/kotlin/com/core/base/corebase/config/AuthenticationFacade.kt
package com.core.base.corebase.config

import com.core.base.corebase.common.exception.BaseException
import com.core.base.corebase.common.code.ErrorCode
import org.springframework.stereotype.Component
import org.springframework.web.context.annotation.RequestScope
import java.util.*

@Component
@RequestScope
class AuthenticationFacade {

    private lateinit var _uid: UUID
    private lateinit var _email: String
    private lateinit var _name: String

    private var flag: Boolean = false

    val uid: UUID by lazy { if (flag) _uid else throw BaseException(ErrorCode.INVALID_TOKEN) }
    val email: String by lazy { if (flag) _email else throw BaseException(ErrorCode.INVALID_TOKEN) }
    val name: String by lazy { if (flag) _name else throw BaseException(ErrorCode.INVALID_TOKEN) }

    fun setInfo(uid: UUID, email: String, name: String) {
        this._uid = uid
        this._email = email
        this._name = name
        this.flag = true
    }

}